plugins/http_plugin/http_plugin.cpp

```diff
@@ -319,62 +549,15 @@ namespace eosio {
                }
 
                con->append_header( "Content-type", "application/json" );
+               con->defer_http_response();
```

Comment:
With this set here, you need to make sure all exits from function call `send_http_response`. If no url_handler found it is not called nor is it called on exception.